pkg/controller/clone/planner.go

```diff
@@ -126,9 +132,10 @@ func (p *Planner) AddCoreWatches(log logr.Logger) error {
 
 // ChooseStrategyArgs are args for ChooseStrategy function
 type ChooseStrategyArgs struct {
-	Log         logr.Logger
-	TargetClaim *corev1.PersistentVolumeClaim
-	DataSource  *cdiv1.VolumeCloneSource
+	Log            logr.Logger
+	TargetClaim    *corev1.PersistentVolumeClaim
+	DataSource     *cdiv1.VolumeCloneSource
+	FallbackReason string
```

Comment:
I think it would be better to have `ChooseStrategy` return a struct that that the strategy and fallbackreason

Reply:
Something like
```
type ChooseStrategyResult struct{
    Strategy cdiv1.CDICloneStrategy
    FallbackReason *string
}
```

Reply:
yep, smells better